launcher/minecraft/launch/LauncherPartLaunch.cpp

```diff
@@ -159,6 +161,113 @@ void LauncherPartLaunch::executeTask()
+    auto canSandbox = false;
+
+#ifdef Q_OS_LINUX
+    const auto bwrapPath = MangoHud::getBwrapBinary();
+    canSandbox = !bwrapPath.isEmpty();
+#endif
+
+    if (wantSandbox && !canSandbox) {
+        const char *reason = QT_TR_NOOP("Sandboxing was requested, but is NOT available on your system.\nPlease turn off sandboxing to proceed launching.");
+        emit logLine(tr(reason), MessageLevel::Error);
+        emitFailed(tr(reason));
+        return;
+    }
+
+#ifdef Q_OS_LINUX
+    if (wantSandbox && canSandbox) {
+        QString actualRuntimeDir = QStandardPaths::writableLocation(QStandardPaths::RuntimeLocation);
+        QString sandboxedRuntimeDir = "/tmp";
+
+        const static QStringList systemBinds{
+            "/etc",
+            "/usr",
+            "/bin",
+            "/sbin",
+            "/lib",
+            "/lib32",
+            "/lib64",
+            "/sys/class",
+            "/sys/dev/char",
+            "/sys/devices/pci0000:00",
+            "/sys/devices/system/cpu",
+            "/run/systemd/resolve",
+        };
+
+        QStringList bwrapArgs{};
+        bwrapArgs << bwrapPath; // will either be taken out using takeFirst or passed to wrapper command
+        bwrapArgs << "--unshare-all";
+        bwrapArgs << "--share-net";
+        bwrapArgs << "--die-with-parent";
+        bwrapArgs << "--unsetenv" << "DBUS_SESSION_BUS_ADDRESS";
+
+        // default binds
+        bwrapArgs << "--dev" << "/dev";
+        bwrapArgs << "--dev-bind-try" << "/dev/dri" << "/dev/dri";
+        bwrapArgs << "--proc" << "/proc";
+        bwrapArgs << "--setenv" << "XDG_RUNTIME_DIR" << sandboxedRuntimeDir;
+
+        for (auto path : systemBinds) {
+            bwrapArgs << "--ro-bind-try" << path << path;
+        }
+
+        // desktop integration
+        bwrapArgs << "--ro-bind-try" << QString("%1/pulse").arg(actualRuntimeDir) << QString("%1/pulse").arg(sandboxedRuntimeDir);
+        bwrapArgs << "--ro-bind-try" << QString("%1/pipewire-0").arg(actualRuntimeDir) << QString("%1/pipewire-0").arg(sandboxedRuntimeDir);
+        {
+            auto display = qEnvironmentVariable("DISPLAY");
+            auto xAuthPath = qEnvironmentVariable("XAUTHORITY",
+                    FS::PathCombine(QStandardPaths::writableLocation(QStandardPaths::HomeLocation), ".Xauthority"));
+            auto wlDisplay = qEnvironmentVariable("WAYLAND_DISPLAY");
+
+            if (display.startsWith(':')) {
+                auto x11Socket = QString("/tmp/.X11-unix/X%1").arg(display.mid(1));
+                bwrapArgs << "--ro-bind-try" << x11Socket << x11Socket;
+            }
+
+            bwrapArgs << "--ro-bind-try" << xAuthPath << xAuthPath;
+
+            if (wlDisplay.startsWith('/'))
+                bwrapArgs << "--ro-bind-try" << wlDisplay << wlDisplay;
+            else
+                bwrapArgs << "--ro-bind-try" << FS::PathCombine(actualRuntimeDir, wlDisplay) << FS::PathCombine(sandboxedRuntimeDir, wlDisplay);
+        }
+
```

Comment:
According to https://wayland.freedesktop.org/docs/html/apb.html#Client-classwl__display_1af048371dfef7577bd39a3c04b78d0374, if there's no WAYLAND_DISPLAY it's still possible that the wayland socket exists, at wayland-0 from the XDG_RUNTIME_DIR